packages/core/src/Ogmios/ogmiosToCore/ogmiosToCore.ts

```diff
@@ -71,17 +74,18 @@ const getBlockAndKind = (block: Ogmios.Schema.Block): BlockAndKind | null => {
   }
 };
 
-const getBlockBodyHash = (b: BlockAndKind): Cardano.BlockId => {
+// REVIEW: I am not sure if this is actually correct
+export const getSlotLeader = (b: BlockAndKind): Cardano.SlotLeader | undefined => {
   switch (b.kind) {
+    case 'babbage':
     case 'byron': {
-      return Cardano.BlockId(b.block.hash);
+      return; // REVIEW: byron blocks do not seem to vrf
     }
     case 'allegra':
     case 'alonzo':
-    case 'babbage':
     case 'mary':
     case 'shelley': {
-      return Cardano.BlockId(b.block.header.blockHash);
+      return Cardano.SlotLeader(b.block.header.leaderValue.proof || '');
```

Comment:
This will throw with empty string. Do you know if this can actually be null at Shelley+ @rhyslbw ?